### ISSUE TITLE
function-signature does not respect max_line_length

### ISSUE THREAD

**Issue by jakoss** [pinterest/ktlint#2296]
## Expected Behavior

```kotlin
override fun onViewCreated(view: View, savedInstanceState: Bundle?) { 
    super.onViewCreated(view, savedInstanceState) 
} 
```

My max_line_length is `180`, i tried to turn off `ktlint_function_signature_wrapping_rule_always_with_minimum_parameters`. As far as i understand this should leave my line intact, but **ktlint seems to always force multiline parameters when i have more than 1 parameter. It works correctly for a single parameter.**. My `.editorconfig` is further down.

## Observed Behavior

```kotlin
override fun onViewCreated( 
	view: View, 
    savedInstanceState: Bundle?, 
) {  
   super.onViewCreated(view, savedInstanceState) 
} 
```

## Steps to Reproduce

Running the ktlint 1.0.0 from CLI on given code should give the observed behavior.

## Your Environment

* Version of ktlint used: 1.0.0
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): CLI
* Version of Gradle used (if applicable): run from CLI
* Operating System and version: MacOS Sonoma

My .editorconfig properties related to ktlint:
```
[*]
charset = utf-8
end_of_line = crlf
indent_size = 4
indent_style = space
insert_final_newline = true
max_line_length = 180
tab_width = 4
trim_trailing_whitespace = true

[{*.kt,*.kts}]
ktlint_standard_no-wildcard-imports = disabled
ktlint_standard_no-empty-class-body = disabled
ktlint_standard_import-ordering = disabled
# Necessary for compatibility with Compose
ktlint_standard_function-naming = disabled
ktlint_standard_annotation = disabled

ktlint_function_signature_wrapping_rule_always_with_minimum_parameters = disabled
```

Log from ktlint CLI:
```
MessageFragment.kt:29:32: Newline expected after opening parenthesis (standard:function-signature)
MessageFragment.kt:29:44: Parameter should start on a newline (standard:function-signature)
MessageFragment.kt:29:71: Newline expected before closing parenthesis (standard:function-signature)
```

**Comment by jakoss** [comment]
@paul-dingemans It is literally there, just in the link to rule documentation you posted:
![image](https://github.com/pinterest/ktlint/assets/3933348/f57ef82c-1399-4eae-bea5-9c3f24bb84ce)

And even ignoring the parameter itself - the issue still persist. Based on the examples from the aforementioned doc suggest that my expected behavior does not happen for some reason

**Comment by paul-dingemans** [comment]
> @paul-dingemans It is literally there, just in the link to rule documentation you posted:

Ow, that is pretty embarassing. Let me reopen this issue until docs are fixed.
